docs/flink-integration-guide.md

```diff
@@ -0,0 +1,219 @@
+
+## Usage Example Code
+
+  Writing flink data to local carbon table.
+
+  ```scala
+    import java.util.Properties
+    import org.apache.carbon.flink.CarbonLocalProperty
+    import org.apache.carbon.flink.CarbonWriterFactory
+    import org.apache.carbon.flink.ProxyFileSystem
+    import org.apache.carbondata.core.constants.CarbonCommonConstants
+    import org.apache.flink.api.common.restartstrategy.RestartStrategies
+    import org.apache.flink.core.fs.Path
+    import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
+    import org.apache.flink.streaming.api.functions.sink.filesystem.StreamingFileSink
+    import org.apache.flink.streaming.api.functions.source.SourceFunction
+
+    val databaseName = "default"
+    val tableName = "test"
+    val tablePath = "/data/warehouse/test"
+    val dataTempPath = "/data/temp/"
+
+    val tableProperties = new Properties
+
+    val writerProperties = new Properties
+    writerProperties.setProperty(CarbonLocalProperty.DATA_TEMP_PATH, dataTempPath)
+
+    val carbonProperties = new Properties
+    carbonProperties.setProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT, CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT)
+    carbonProperties.setProperty(CarbonCommonConstants.CARBON_DATE_FORMAT, CarbonCommonConstants.CARBON_DATE_DEFAULT_FORMAT)
+    carbonProperties.setProperty(CarbonCommonConstants.UNSAFE_WORKING_MEMORY_IN_MB, "1024")
+
+    val writerFactory = CarbonWriterFactory.builder("Local").build(
+      databaseName,
+      tableName,
+      tablePath,
+      tableProperties,
+      writerProperties,
+      carbonProperties
+    )
+
+    val environment = StreamExecutionEnvironment.getExecutionEnvironment
+    environment.setParallelism(1)
+    environment.enableCheckpointing(2000L)
+    environment.setRestartStrategy(RestartStrategies.noRestart)
+
+    // Define a custom source.
+    val source = new SourceFunction[Array[AnyRef]]() {
+      override
+      def run(sourceContext: SourceFunction.SourceContext[Array[AnyRef]]): Unit = {
+        // Array length should equals to table column count, and values order in array should matches table column order.
+        val data = new Array[AnyRef](3)
+        data(0) = "value1"
+        data(1) = "value2"
+        data(2) = "value3"
+        sourceContext.collect(data)
+      }
+
+      override 
+      def cancel(): Unit = {
+        // do something.
+      }
+    }
+
+    val stream = environment.addSource(source)
+    val streamSink = StreamingFileSink.forBulkFormat(new Path(ProxyFileSystem.DEFAULT_URI), writerFactory).build
+
+    stream.addSink(streamSink)
+
+    try {
+      environment.execute
+    } catch {
+      case exception: Exception =>
+        // TODO
+        throw new UnsupportedOperationException(exception)
+    }
+  ```
+
+  Insert into table from stage directory.
+  
+  ```sql
+    INSERT INTO test STAGE
```

Comment:
Can create a table test to define the scenario

Reply:
OK